pkg/cache/tas_cache_test.go

```diff
@@ -709,6 +712,266 @@ func TestFindTopologyAssignment(t *testing.T) {
+						},
+					},
+				},
+			},
+			pods: []corev1.Pod{
+				*testingpod.MakePod("test-unscheduled", "test-ns").
+					Request(corev1.ResourceCPU, "600m").
+					Obj(),
+			},
+			request: kueue.PodSetTopologyRequest{
+				Required: ptr.To(tasHostLabel),
+			},
+			levels: defaultOneLevel,
+			requests: resources.Requests{
+				corev1.ResourceCPU: 600,
+			},
+			count: 1,
+			wantAssignment: &kueue.TopologyAssignment{
+				Levels: defaultOneLevel,
+				Domains: []kueue.TopologyDomainAssignment{
+					{
+						Count: 1,
+						Values: []string{
+							"x1",
+						},
+					},
+				},
+			},
+		},
+		"don't consider terminal pods when computing the capacity": {
+			nodes: []corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "x1",
+						Labels: map[string]string{
+							tasHostLabel: "x1",
+						},
+					},
+					Status: corev1.NodeStatus{
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("1"),
+							corev1.ResourceMemory: resource.MustParse("1Gi"),
+						},
+					},
+				},
+			},
+			pods: []corev1.Pod{
+				*testingpod.MakePod("test-failed", "test-ns").NodeName("x1").
+					Request(corev1.ResourceCPU, "600m").
+					StatusPhase(corev1.PodFailed).
+					Obj(),
+				*testingpod.MakePod("test-succeeded", "test-ns").NodeName("x1").
+					Request(corev1.ResourceCPU, "600m").
+					StatusPhase(corev1.PodSucceeded).
+					Obj(),
+			},
+			request: kueue.PodSetTopologyRequest{
+				Required: ptr.To(tasHostLabel),
+			},
+			levels: defaultOneLevel,
+			requests: resources.Requests{
+				corev1.ResourceCPU: 600,
+			},
+			count: 1,
+			wantAssignment: &kueue.TopologyAssignment{
+				Levels: defaultOneLevel,
+				Domains: []kueue.TopologyDomainAssignment{
+					{
+						Count: 1,
+						Values: []string{
+							"x1",
+						},
+					},
+				},
+			},
+		},
+		"don't consider TAS pods when computing the capacity": {
+			nodes: []corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "x1",
+						Labels: map[string]string{
+							tasHostLabel: "x1",
+						},
+					},
+					Status: corev1.NodeStatus{
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("1"),
+							corev1.ResourceMemory: resource.MustParse("1Gi"),
+						},
+					},
+				},
+			},
+			pods: []corev1.Pod{
+				*testingpod.MakePod("test-tas", "test-ns").NodeName("x1").
+					Request(corev1.ResourceCPU, "600m").
+					Label(kueuealpha.TASLabel, "true").
+					StatusPhase(corev1.PodRunning).
+					Obj(),
+			},
+			request: kueue.PodSetTopologyRequest{
+				Required: ptr.To(tasHostLabel),
+			},
+			levels: defaultOneLevel,
+			requests: resources.Requests{
+				corev1.ResourceCPU: 600,
+			},
+			count: 1,
+			wantAssignment: &kueue.TopologyAssignment{
+				Levels: defaultOneLevel,
+				Domains: []kueue.TopologyDomainAssignment{
+					{
+						Count: 1,
+						Values: []string{
+							"x1",
+						},
+					},
+				},
+			},
+		},
```

Comment:
One way to fix that would be to populate `TASFlavorCache.usage` with resources allocated for TAS pods

Reply:
You are right, though TAS capacity is tracked on prod based on Workloads rather than pods.
Still, probably better that way, I have adjusted the test case code and added a comment.
I think ideally we have this covered at the integration level.

Reply:
I think a scenario like this (not double counting usage) we could also cover with e2e tests. Basically create 2 workloads that fit. It will not be that easy in integration tests because integration tests don't create pods. I have started a suite in another PR so ones it is merged I think I could extend it .